llvm/lib/Analysis/LoopAccessAnalysis.cpp

```diff
@@ -1454,6 +1464,9 @@ static bool isNoWrapAddRec(Value *Ptr, const SCEVAddRecExpr *AR,
   if (AR->getNoWrapFlags(SCEV::NoWrapMask))
     return true;
```

Comment:
I wonder off-hand if this is ever hit?

Reply:
Also, the API isn't very clear, as AR is only ever used here, and what's actually checked is the Ptr, not AR. Perhaps the function should be renamed to isNoWrapGEP, and the AR check can be moved to the caller, assuming that it's useful?

Reply:
Sounds good to me